ompi/mca/topo/treematch/topo_treematch_dist_graph_create.c

```diff
@@ -683,7 +683,7 @@ int mca_topo_treematch_dist_graph_create(mca_topo_base_module_t* topo_module,
                 free(aff_mat->sum_row);
                 free(aff_mat);
                 tm_free_solution(sol);
-                tm_free_tree(comm_tree);
+                free(comm_tree);
```

Comment:
This structure has been allocated with treematch functions, I think it is safer to release it with treematch functions. From a programming perspective the arguments should have been ** as the structure is really freed, but the structure is correctly released (except for "dumb" trees in free_non_constraint_tree).

@GuillaumeMercier can you clarify this ?